AIAgent/ml/training/validation.py

```diff
@@ -82,11 +84,16 @@ def validate_coverage(
             if isinstance(result, GameError):
                 need_to_save_map: bool = result.need_to_save_map()
                 if not need_to_save_map:
-                    statistics_collector.fail(result._map)
+                    if not fail_immediately:
+                        statistics_collector.fail(result._map)
+                    else:
+                        failed_validation = True
+                        break
```

Comment:
We should try run all maps, but fail at the end if at least one failed.

Reply:
https://github.com/PySymGym/PySymGym/pull/104/commits/e8b533cc9b54468fad767cee26916a9758829a4d